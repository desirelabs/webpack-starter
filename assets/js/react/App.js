import React from 'react'

import Button from './Button'
import Tick from './Tick'

class App extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      name: '',
      people: {},
      count: 0
    }

    this.setData = this.setData.bind(this)
  }

  isOdd (value) {
    return value % 2 === 0
  }

  componentWillMount () {
    this.setData()
  }

  setData () {
    this.setState({
      name: 'React'
    })
  }

  add () {
    this.setState({
      count: this.state.count + 1
    })
  }

  remove () {
    if (this.state.count > 0) {
      this.setState({
        count: this.state.count - 1
      })
    }
  }

  render () {
    return (
      <div>
        <h1>Hello {this.state.name}!</h1>
        <Tick count={this.state.count} />
        <Button text={'Ajouter'} actionClick={() => this.add()} />
        <Button text={'Supprimer'} actionClick={() => this.remove()} />
      </div>
    )
  }
}
// eslint-disable-next-line
const style = {
  display: 'flex',
  flexFlow: 'column nowrap',
  alignItems: 'center'
}

export default App
