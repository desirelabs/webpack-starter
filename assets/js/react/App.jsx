import React from 'react'

class App extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      name: '',
      people: {}
    }

    this.setData = this.setData.bind(this)
  }

  componentWillMount () {
    this.setData()
  }

  setData () {
    this.setState({
      name: 'React'
    })
  }

  render () {
    return (
      <h1>Hello {this.state.name}!</h1>
    )
  }
}

export default App
