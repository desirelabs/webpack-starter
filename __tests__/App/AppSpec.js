
import React from 'react'
import App from '../../assets/js/react/App'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount, shallow } from 'enzyme'

configure({ adapter: new Adapter() })

describe('App', () => {
  /**
   * Component's boilerplate
   */
  let props
  let mountedApp
  const app = () => {
    if (mountedApp) {
      mountedApp = mount(<App {...props} />)
    }
    return mountedApp
  }
  beforeEach(() => {
    props = undefined
    mountedApp = undefined
  })

  /**
   * Snapshot testing
   */
  test('Should render', () => {
    const wrapper = renderer.create(<App />).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  /**
   * Html testing
   */
  test('Always render a title tag h1', () => {
    const wrapper = shallow(<App />)
    const title = wrapper.find('h1')
    expect(title).toHaveLength(1)
  })

  /**
   * Html content text testing
   */
  test('Expects title tag content to exist', () => {
    const wrapper = shallow(<App />)
    const title = wrapper.find('h1')
    expect(title.text().length).toBeGreaterThan(0)
  })

  /**
   * String contains
   */
  test('Expects title tag content to contain `Hello`', () => {
    const wrapper = shallow(<App />)
    const title = wrapper.find('h1')
    expect(title.text()).toEqual(expect.stringContaining('Hello'))
  })

  /**
   * Html string from state testing
   */
  test('Expects title tag content to contain `React`', () => {
    const wrapper = shallow(<App />)
    const title = wrapper.find('h1')
    expect(title.text().indexOf('React')).toBeGreaterThan(-1)
  })

  /**
   * Testing state property value
   */
  test('Should update state', () => {
    const wrapper = mount(<App />)
    wrapper.instance().setData()
    expect(wrapper.state('name')).toEqual('React')
  })

  /**
   * Testing component's method
   */
  test('Should be odd', () => {
    const wrapper = mount(<App />)
    expect(wrapper.instance().isOdd(2)).toBeTruthy()
  })

  /**
   * Testing array to contain provided values
   */
  const array = ['Alice', 'Bob']
  test('Should array to contain all expected values', () => {
    expect(['Alice', 'Bob', 'Eve']).toEqual(expect.arrayContaining(array))
  })

  /**
   * Same as before with NOT
   */
  test('does not match if received does not contain expected elements', () => {
    expect(['Bob', 'Eve']).not.toEqual(expect.arrayContaining(array))
  })

  /**
   * Testing object to contain provided values
   */
  const object = {name: 'toto'}
  test('Should Object to contain all expected values', () => {
    expect({name: 'toto', age: 12}).toEqual(expect.objectContaining(object))
  })

  /**
   * Same as before with NOT
   */
  test('does not match if received does not contain expected elements', () => {
    expect({name: 'poney'}).not.toEqual(expect.objectContaining(object))
  })
})
