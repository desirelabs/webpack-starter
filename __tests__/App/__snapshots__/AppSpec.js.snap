// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render 1`] = `
<div>
  <h1>
    Hello 
    React
    !
  </h1>
  <div>
    Ticks : 
    0
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      
    </button>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      
    </button>
  </div>
</div>
`;
